shell/browser/api/electron_api_safe_storage.cc

```diff
@@ -31,12 +31,24 @@ void SetElectronCryptoReady(bool ready) {
 #endif
 
 bool IsEncryptionAvailable() {
+#if BUILDFLAG(IS_LINUX)
+  // Calling IsEncryptionAvailable() before the app is ready results in a crash
+  // on Linux.
+  // Refs: https://github.com/electron/electron/issues/32206.
+  if (!Browser::Get()->is_ready())
+    return false;
+#endif
   return OSCrypt::IsEncryptionAvailable();
 }
 
 v8::Local<v8::Value> EncryptString(v8::Isolate* isolate,
                                    const std::string& plaintext) {
-  if (!OSCrypt::IsEncryptionAvailable()) {
+  if (!IsEncryptionAvailable()) {
+    if (!Browser::Get()->is_ready()) {
+      gin_helper::ErrorThrower(isolate).ThrowError(
+          "safeStorage cannot be used before app is ready");
+      return v8::Local<v8::Value>();
+    }
```

Comment:
I think it's a bit clearer to put this check first, before the `IsEncryptionAvailable` check. After that, you can remove the `IsEncryptionAvailable` helper and replace calls with a direct call to `OSCrypt::IsEncryptionAvailable`.

Reply:
> put this check first, before the IsEncryptionAvailable check

But we can still access the safeStorage API before the app is ready on macOS

> remove the IsEncryptionAvailable helper

We won't be able to expose `OSCrypt::IsEncryptionAvailable()` directly from `electron_browser_safe_storage` because direct calls would crash on Linux before the app is ready.

Reply:
> we can still access the safeStorage API before the app is ready on macOS

let's make this consistent on all platforms. Even though it's technically possible to call it before ready on macOS, if an app does that and the developer tests on mac then ships to linux, their app will fail there. Keeping it consistent on all platforms makes things more reliable.

Reply:
> let's make this consistent on all platforms.

Since that would be a breaking change for macOS, maybe we could do that in a follow up PR, so that it's possible to at least fix this crash in all supported release lines?

Reply:
It's debatable that it's a breaking change; I'd be okay classifying it as a bugfix—it was never explicitly documented that this worked before app.on('ready')! But I don't feel strongly about it and I'm fine with doing the consistency work in a followup.